src/Microsoft.OData.Core/ODataInstanceAnnotation.cs

```diff
@@ -62,7 +62,9 @@ internal ODataInstanceAnnotation(string annotationName, ODataValue annotationVal
         /// <param name="name">Name to validate.</param>
         internal static void ValidateName(string name)
         {
-            if (name.IndexOf('.') < 0 || name[0] == '.' || name[name.Length - 1] == '.')
+            // being internal, all callers validate that the name is not empty, but don't validate anything else; here, we validate that there is a namespace, 
+            // indicated by the presence of a period and by characters before and after the period
+            if (name.Length < 3 || name[0] == '.' || name[name.Length - 1] == '.' || name.IndexOf('.', 1, name.Length - 2) < 0)

```

Comment:
@corranrogue9 In the absence of any data to support this optimization, it could easily cause a regression. In my experience, the most common violation is the absence of a `.`. It's likely going to be a leading or a trailing `.`. Which means we might end up introducing 3 comparisons from uncommon/rare scenarios before the comparison for the more common/likely scenario.